src/backend/distributed/commands/utility_hook.c

```diff
@@ -466,9 +466,35 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 			activeDropSchemaOrDBs++;
 		}
 
+		/*
+		 * Check if we are running ALTER EXTENSION citus UPDATE (TO "<version>") command and
+		 * the available version is different than the current version of Citus. In this case,
+		 * ALTER EXTENSION citus UPDATE command can actually update Citus to a new version.
+		 */
+		bool isAlterExtensionUpdateCitusStmt = isCreateAlterExtensionUpdateCitusStmt &&
+											   IsA(parsetree, AlterExtensionStmt);
+
+		bool citusCanBeUpdatedToAvailableVersion = false;
+
+		if (isAlterExtensionUpdateCitusStmt)
+		{
+			citusCanBeUpdatedToAvailableVersion = !InstalledAndAvailableVersionsSame();
+		}
+
 		standard_ProcessUtility(pstmt, queryString, context,
 								params, queryEnv, dest, completionTag);
 
+		/*
+		 * if we are running ALTER EXTENSION citus UPDATE (to "<version>") command, we may need
+		 * to mark existing objects as distributed depending on the "version" parameter if
+		 * specified in "ALTER EXTENSION citus UPDATE" command
+		 */
+		if (isAlterExtensionUpdateCitusStmt)
```

Comment:
It would actually be slightly cleaner to just include `&& citusCanBeUpdatedToAvailableVersion` instead of passing it as an argument.